internal/transport/transport.go

```diff
@@ -717,9 +731,15 @@ type ServerTransport interface {
 	// handlers will be terminated asynchronously.
 	Close(err error)
 
+	// LocalAddr returns the local network address.
+	LocalAddr() net.Addr
+
 	// RemoteAddr returns the remote network address.
 	RemoteAddr() net.Addr
 
+	// Peer returns the peer of the server transport.
+	Peer() *peer.Peer
```

Comment:
Should we delete Local/RemoteAddr and only have this?

Reply:
Sure, done. There's an in flight PR for this but the contributor isn't getting back to my comments so I'll go ahead and just add it to this PR.